vm/interpreter/src/default_runtime.rs

```diff
@@ -265,25 +270,25 @@ where
         };
         self.caller_validated = false;
 
-        // snapshot state tree
-        let snapshot = self
-            .state
-            .snapshot()
-            .map_err(|e| actor_error!(fatal("failed to create snapshot {}", e)))?;
-
         let send_res = vm_send::<BS, SYS, R, P>(self, &msg, None);
 
         // Reset values back to their values before the call
         self.vm_msg = prev_msg;
         self.caller_validated = prev_val;
 
-        send_res.map_err(|e| {
-            if let Err(e) = self.state.revert_to_snapshot(&snapshot) {
+        let ret = send_res.map_err(|e| {
+            if let Err(e) = self.state.revert_to_snapshot() {
                 actor_error!(fatal("failed to revert snapshot: {}", e))
             } else {
                 e
             }
-        })
+        });
+        if let Err(e) = self.state.clear_snapshot() {
+            actor_error!(fatal("failed to clear snapshot: {}", e));
+        }
+        let ret = ret?;
+
+        Ok(ret)
```

Comment:
haha, `let ret = ret?` is so funny. 

You can remove that and just do Ok(ret?)

Reply:
nah, that would be a bug. The reason you need to have it after the clear, is that returning an error here may not be fatal, and the snapshot would not be cleared and run into a bug later